### PR TITLE
Added fastavalidator

### LINKED ISSUE #4427: new module: fastavalidator
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by GallVp**
## PR checklist

Closes #4427 

- [ ] Added new module: fastavalidator.
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/fastavalidator/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                    file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
+                ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match() },
```
Comment (sateeshperi) [inline modules/nf-core/fastavalidator/tests/main.nf.test:27]:
```suggestion
                { assert snapshot(process.out).match("fasta_valid") },
```

File: modules/nf-core/fastavalidator/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                    file(params.test_data['sarscov2']['genome']['genome_gff3'], checkIfExists: true)
+                ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match() },
```
Comment (sateeshperi) [inline modules/nf-core/fastavalidator/tests/main.nf.test:52]:
```suggestion
                { assert snapshot(process.out).match("gff3_invalid") },
```

**Review by sateeshperi — COMMENTED** [review]
if you have multiple tests its better to have unique identifiers in the snap

**Comment by GallVp** [comment]
Thank you for the feedback @sateeshperi 

I was under the impression that by default the snapshot name is the same as the test name. As both the tests have unique names, thus, the snapshots should have unique identifiers. This is documented here: https://code.askimed.com/nf-test/docs/assertions/snapshots/#constructing-complex-snapshots

**Comment by sateeshperi** [comment]
ah you are right! this came in with the latest release....thanks for pointing it out!

also gr8 job with the tests!

**Review by sateeshperi — APPROVED** [review]

**Pull request merged**